### DISCUSSION TITLE
How do you update the HTML?

### DISCUSSION THREAD

**Discussion by jdacoello** [OnToology/OnToology#614] [Q&A] [answered]
Could you please indicate what are the steps to update the HTML?

This is what I tried:
- Modify ontology
- Save, commit, and push changes
  - My repos > Select repo > republish
  - Past jobs > Generate documentation
- Merge pull request 

By searching for any new added information, I can see that the changes are there in some of the files.
However, I can also see that the HTML with the rendered documentation shows the previos version.

**Comment by dgarijo** [comment]
@jdacoello,
every time you modify your ontology, your new changes should be updated in a pull request. 
There is no need to republish your ontology, as that has a different functionality.

If there are still no changes, then there may be a bug. @ahmad88me can you please check when you have the time?

**Comment by ahmad88me** [comment, marked as answer]
@jdacoello Updating the ontology would trigger OnToology to regenerate the resources (e.g., documentation, diagrams, evaluation repo).  OnToology should create a pull requests with these changes. You can accept the pull request if you like the generated resources. 

If the changes in the ontology are not reflected in the documentation then you can reach out to us and we can look further into it. 

* *republish* : would publish the documentation that you already have to the w3id uri that you reserved. It does not change the documentation.

* *Past jobs* : is a readonly view. It shows the progress of what is happening and what happened before. If you refresh you can see the progress bar and the status changing. It is just to see what is going on with your repos on the server.

In short, if you want the new changed in your ontology to be reflected in the documentation, you can do the following:
1. Change the ontology and push the changes to your GitHub.
2. Look at the pull request and accept if you like the newly generated resources.
3. If you want the published documentation to be updated, you can also click on *republish*.

I hope this answers your question. Let us know if you have more questions.

Best,

**Reply by jdacoello** [reply]
Thank you @ahmad88me for the detailed answer. However, it is not working as you described. 
After doing step 1, no PR is generated. The PR is only available when I click on `Past Jobs` > `Ontology` > `Generate documentation` > `main` branch.
I merged the PR (on the main branch). And the website still displays the previous status of the documentation. I can see that nothing has changed in the `gh-pages` branch. Am I missing something?

**Comment by ahmad88me** [comment]
Hi @jdacoello 

1. No PR is generated. I just tested it on my repo and it works. I can't reproduce the issue. Can you check if you have the webhook on your repo. You can do that by going to `settings`-> `Webhooks`. You should be able to see webhook to OnToology (check attached screenshot). 
<img width="746" alt="Screenshot 2022-04-04 at 9 53 57 AM" src="https://user-images.githubusercontent.com/6922709/161505307-0bc69b4d-e3ec-46c4-81de-55a42c80480c.png">
Please try again to change the ontology and push the changes to the main branch. Let us know if the issue persists and we can think about how to reproduce the issue.

2. OnToology does not republish the ontology automatically. To republish your ontology, you can go to `my repositories` -> `Select repo` -> then you can choose the branch -> then click on `republish` for the ontology of choice. 

Let us know if this works for you.
Best,

**Reply by jdacoello** [reply]
Thank you for the quick response. 

I think I messed up the branches because I found files that should not have been in `gh-pages` and viceversa. So, I deleted the `gh-pages`branch and tried again with a clean branch. It seems to be working now. Thanks for your time.
